src/librustc/ty/sty.rs

```diff
@@ -862,23 +863,24 @@ impl<'tcx> PolyFnSig<'tcx> {
 pub struct ParamTy {
     pub idx: u32,
     pub name: Name,
+    pub span: Span,
```

Comment:
We should just add a `DefId` if we want to get back the definition, `Span` is a *very* bad idea within anything reachable from `Ty`.

Reply:
Yes, definitely not a span. I have been resisting the idea of going back to `DefId`, because I'd *rather* we move towards just an index, but that would require having more context about the currently "in scope" parameters somewhere to give good error messages. Maybe a `DefId` is the right step for now.

Reply:
I think we use `DefId` in `Region`, don't we?